src/wasm-traversal.h

```diff
@@ -115,6 +117,196 @@ struct Visitor {
   }
 };
 
+// A visitor which must be overridden for each visitor that is reached.
+
+template<typename SubType, typename ReturnType = void>
+struct OverriddenVisitor {
+  // Expression visitors
+  ReturnType visitBlock(Block* curr) {
```

Comment:
Hm, this is a bit verbose. I wonder if we could turn it into a macro a la DELEGATE?

Reply:
Done.

Reply:
Hmm, but it stopped working (throwing an error).

Reply:
Weird, it looks exactly equivalent except for the extra semicolon (although since I'm nitpicking, it probably shouldn't *actually* be called DELEGATE since it's not delegating, it's just asserting).

Reply:
Looks like static asserts only happen if the code is called. Which is good enough.

Reply:
Yeah, it is the same as before. I just happened to test a method that wasn't called now and not earlier ;)

Maybe delegate isn't the best name, but it's also writing a stub for the function with an abort in it, so it's more than the static assert. Maybe "handle"? or "implement"?

Reply:
how about unhandled/unimplemented?

Reply:
Sure, done.